Retour du Forum PHP 2021 (#281)
BedrockStreaming/tech.bedrockstreaming.com: _posts/2021-11-02-forum-php-2021.md

---
layout: post
title: "Forum PHP 2021 - L'édition des retrouvailles"
description: "Nos retours du Forum PHP à Paris, les 21 et 22 octobre 2021"
author:
    name: Les Bedrockers
avatar:
email:
twitter: Bedrock_Tech
github: 
category:
tags: [afup, php, forumphp, conference]
image:
  feature: posts/forumphp2021/header.jpg
comments: true
language: fr
---

Cette année encore, Bedrock participait au Forum PHP où était proposé une grande diversité de conférences.
Des sujets techniques et d'autres, plus génériques, étaient abordés : Symfony 6, Git, environnement, sous-représentation des femmes dans l'informatique…

> Avec une [Sofia LESCANO](https://twitter.com/SofLesc), [Benoit VIGUIER](https://twitter.com/b_viguier) sur les planches et une quinzaine de participantes et participants dans le publique, l'occasion de rencontrer à nouveau la communauté PHP en chair et en os a été saisie avec une certaine impatience.

Reviewer: ```suggestion
Avec une [Sofia LESCANO](https://twitter.com/SofLesc), un [Benoit VIGUIER](https://twitter.com/b_viguier) sur les planches et une quinzaine de participantes et participants dans le publique, l'occasion de rencontrer à nouveau la communauté PHP en chair et en os a été saisie avec une certaine impatience.
```

> Avec une [Sofia LESCANO](https://twitter.com/SofLesc), [Benoit VIGUIER](https://twitter.com/b_viguier) sur les planches et une quinzaine de participantes et participants dans le publique, l'occasion de rencontrer à nouveau la communauté PHP en chair et en os a été saisie avec une certaine impatience.

Reviewer 2: je trouves pas ça terrible de dire "Avec une", je dirais simplement 
```suggestion
Avec [Sofia LESCANO](https://twitter.com/SofLesc), [Benoit VIGUIER](https://twitter.com/b_viguier) sur les planches et une quinzaine de participantes et participants dans le publique, l'occasion de rencontrer à nouveau la communauté PHP en chair et en os a été saisie avec une certaine impatience.
```

> Plusieurs conférences ont retenu notre attention et auront un impact à court terme sur nos projets :

Reviewer 2: ```suggestion
Plusieurs conférences ont retenu notre attention et auront un impact à court terme sur nos projets :

```

[Anne-Laure DE BOISSIEU](https://twitter.com/AnneLaure2B) et [Amélie DEFRANCE](https://twitter.com/amelie_defrance), dans leur conférence intitulée _"Accessibilité et SEO : et si on relevait le niveau ?"_, ont rappelé, entre autres,  quelques règles fondamentales sur l'accessibilité de nos sites Internet. Fort de ce rappel, nous espérons apporter des améliorations sur l'accessibilité de notre back-office. Par exemple, retravailler le contraste couleur de certains de nos écrans ou encore ajouter du contenu dans nos balises HTML permettra d'améliorer la compréhension.

> [Emeric](https://twitter.com/emerick42) et [Jérémy](https://twitter.com/JamesJrmy) nous expliquent que leurs clients respectifs ont un même besoin : "Déclencher une action automatique à un moment précis, sans aucune limite dans le temps". Un des exemples est "Pouvoir supprimer, automatiquement, un panier d’achat au bout de 15 minutes".

Reviewer 3: ```suggestion
[Emeric](https://twitter.com/emerick42) et [Jérémy](https://twitter.com/JamesJrmy) nous ont expliqué que leurs clients ont un même besoin : _"Déclencher une action automatique à un moment précis, sans aucune limite dans le temps"_. Par exemple : _"supprimer, automatiquement, un panier d’achat au bout de 15 minutes"_.
```

> Après de longues recherches, il s’avère qu’il n’existe rien sur le marché pour un tel besoin. Et nous, BedRock, pouvons confirmer cela car nous avons le même besoin et n’avons rien trouvé.

Reviewer 3: ```suggestion
Après de longues recherches, il n’existe rien sur le marché pour répondre à un tel besoin. Et nous, BedRock, confirmons : nous avons le même besoin et n’avons rien trouvé non plus.
```

> Kairoi est donc né. Kairoi c’est une application serveur Rust de planification de tâche, avec son propre protocole inspiré de celui de Redis qui permet de :

Reviewer 3: ```suggestion
Kairoi est donc né. C’est une application serveur Rust de planification de tâches, avec son propre protocole _(inspiré de celui de Redis)_ qui permet de :
```

* Récupérer des évènements à planifier

> * Connaître l’état de l’évènement

Reviewer 3: ```suggestion
* Connaître l’état d’un évènement
```

> * Déclencher l'événement au moment opportun

Reviewer 3: ```suggestion
* Le déclencher au moment opportun
```

  * soit sur un protocole AMQP
  * soit sur un shell
  

> Le Forum PHP est l'occasion de parler de sujets pointus techniquement mais aussi une occasion d'échange et de partage autour de sujets plus transversaux.

Reviewer: ```suggestion
Le Forum PHP est l'occasion de parler de sujets pointus techniquement, mais aussi une occasion d'échange et de partage autour de sujets plus transversaux.
```

> Notre domaine, l'IT, comme bien d'autres, est sensible au sujet de l'écologie. Deux axes de réflexion ont été évoqués aux travers deux conférences.

Reviewer 3: ```suggestion
Notre domaine, l'IT, comme bien d'autres, est sensible au sujet de l'écologie. Deux axes de réflexion ont été évoqués pendant deux conférences.
```

> En ouverture, [François Zaninotto](https://twitter.com/francoisz) partage avec entrain son questionnement sur sa recherche de mesures fiables des dépenses énergétiques. Cette recherche l'a mené finalement à développer  [GreenFrame](https://greenframe.io/), un outil en cours de construction chez Marmelab qui a pour but de cibler au plus près la consommation énergétique afin de tendre _"Vers la sobriété numérique"_.

Reviewer 3: "partage" : dans certains paragraphes à propos de certain talks, vous avez écrit au passé ; dans d'autres (dont ici), au présent. Uniformiser ?

> En ouverture, [François Zaninotto](https://twitter.com/francoisz) partage avec entrain son questionnement sur sa recherche de mesures fiables des dépenses énergétiques. Cette recherche l'a mené finalement à développer  [GreenFrame](https://greenframe.io/), un outil en cours de construction chez Marmelab qui a pour but de cibler au plus près la consommation énergétique afin de tendre _"Vers la sobriété numérique"_.

Reviewer 3: ```suggestion
En ouverture, [François Zaninotto](https://twitter.com/francoisz) partage avec entrain sa recherche de mesures fiables des dépenses énergétiques. Elle l'a menée à développer  [GreenFrame](https://greenframe.io/), un outil en cours de construction chez Marmelab, qui a pour but de cibler au plus près la consommation énergétique afin de tendre _"Vers la sobriété numérique"_.
```

Reviewer 3: "Elle l'a mené", ou "Elle l'a menée" ? Je ne sais jamais :-/
Antidote a l'air de dire qu'il faut un e ?
![Capture d’écran 2021-11-02 à 10 07 56](https://user-images.githubusercontent.com/193481/139817882-a74fbf71-b7de-44a6-a1de-30e36dcbbae2.png)

> [Hélène MAITRE-MARCHOIS](https://twitter.com/helenemaitre) a sû mettre en perspective le rôle de chaque développeuse et développeur en insistant sur le fait que la responsabilité du dérèglement climatique n'est pas forcément là où on l'attend. Avec _"Comment sauver la planète en ne faisant rien"_, elle entend faire prendre conscience que si, la production et consommation de contenu représente des pôles sur lesquels en tant que tech, nous pouvons agir. Le renouvellement du parc reste une cause prépondérante dans l'impact écologique.

Reviewer 3: "la production et consommation de contenu représente des pôles"

"la production et consommation" ça pourrait être singulier ou pluriel.
Mais "des pôles" est pluriel.
Donc "représentent", je pense ?

Le renouvellement accéléré par le foisonnement de nouvelles fonctionnalités, trop souvent inutiles, est une obsolescence programmée.

> Utile, Accessible, Durable. - Voilà trois notions simples qui pourtant peuvent nous permettre de faire la différence.

Reviewer: ```suggestion
Utile, Accessible, Durable. - Voilà trois notions simples qui peuvent pourtant nous permettre de faire la différence.
```

> Utile, Accessible, Durable. - Voilà trois notions simples qui pourtant peuvent nous permettre de faire la différence.

Reviewer 3: ```suggestion
Utile, Accessible, Durable. Voilà trois notions simples qui, pourtant, peuvent nous permettre de faire la différence.
```

> [Nicolas GREKAS](https://twitter.com/nicolasgrekas), principal engineer Symfony, est venu nous parler de l'écosystème de ce framework. Il nous a présenté le [calendrier de livraisons](https://symfony.com/releases) et de maintenance des différentes versions, avec la sortie d'une nouvelle majeure prévue tous les deux ans. Chaque majeure voit la suppression du code déprécié dans la version précédente. Symfony 6 par exemple est un Symfony 5.4 sans les dépréciations. Les dernières versions avant une majeure (la 4.4 ou 5.4 par exemple) sont assurées d'avoir un support à long terme.

Reviewer 3: ```suggestion
[Nicolas GREKAS](https://twitter.com/nicolasgrekas), principal engineer Symfony, nous a parlé de l'écosystème de ce framework. Il nous a présenté le [calendrier de livraisons](https://symfony.com/releases) et de maintenance des différentes versions, avec la sortie d'une nouvelle majeure prévue tous les deux ans. Chaque majeure voit la suppression du code déprécié dans la version précédente. Par exemple, Symfony 6 est un Symfony 5.4 sans ses dépréciations. Les dernières versions avant une majeure (comme la 4.4 ou 5.4) sont assurées d'avoir un support à long terme.
```

Reviewer 3: "support à long terme", ça serait cool de préciser ce que ça veut dire (quelle durée, notamment)

> Concernant le développement des prochaines versions de Symfony, l'accent a été mis sur la compatibilité avec PHP8. La majorité du travail consistant au remplacement des annotations @return par un typage natif, Nicolas en a profité pour nous parler d'un outil nommé [patch-type-declarations](https://symfony.com/doc/5.4/setup/upgrade_major.html#upgrading-to-symfony-6-add-native-return-types), permettant d'automatiser cette tâche.

Reviewer 3: ```suggestion
Pour les prochaines versions de Symfony, l'accent est mis sur la compatibilité avec PHP8. Puisque la majorité du travail consiste à remplacer les annotations `@return` par un typage natif, Nicolas a parlé de l'outil [patch-type-declarations](https://symfony.com/doc/5.4/setup/upgrade_major.html#upgrading-to-symfony-6-add-native-return-types) qui automatise cette tâche.
```

> Et pour finir cette série de conférences nous avons suivi l'incroyable histoire de WorkAdventure ( _"WorkAdventure de la genèse à aujourd'hui : Retour d'expérience sur 1 an d'univers virtuels"_ - [David NÉGRIER](https://twitter.com/david_negrier)), le résultat d'un hackathon fait pour pallier l'ennui des confinements qui est devenu le support d'événements majeurs de l'année dernière.

Reviewer 3: ```suggestion
Pour finir cette série de conférences, nous avons suivi l'incroyable histoire de WorkAdventure, lors de _"WorkAdventure de la genèse à aujourd'hui : Retour d'expérience sur 1 an d'univers virtuels"_ présenté par [David NÉGRIER](https://twitter.com/david_negrier)) : le résultat d'un hackathon fait pour pallier l'ennui des confinements, qui est devenu le support d'événements majeurs l'année dernière.
```

> ![Photo de toute l'équipe Bedrock](/images/posts/forumphp2021/bedrock_team.jpg)

Reviewer 3: Il ne s'agit pas de "toute" l'équipe ;-)

## Les speakers Bedrock

> Lors de cette édition, deux Bedrockers ont eu l'opportunité de présenter un sujet, l'occasion pour nous de demander à [Sofia](https://twitter.com/SofLesc) _"Faites confiance aux développeurs.euses de votre équipe : voyez plus loin que les fonctionnalités"_ et [Benoit](https://twitter.com/b_viguier) _"Fiber : la porte ouverte sur l'asynchrone"_ comment ils ont vécu cet événement.

Reviewer: ```suggestion
Lors de cette édition, deux Bedrockers ont eu l'opportunité de présenter un sujet, l'occasion pour nous de demander à [Sofia Lescano](https://twitter.com/SofLesc) _"Faites confiance aux développeurs.euses de votre équipe : voyez plus loin que les fonctionnalités"_ et [Benoit Viguier](https://twitter.com/b_viguier) _"Fiber : la porte ouverte sur l'asynchrone"_ comment ils ont vécu cet événement.
```

> **_Comment vous est venu l'idée de soumettre un sujet de conférence, et comment avez-vous abordé sa préparation ?_**

Reviewer 3: ```suggestion
**_Comment vous est venue l'idée de soumettre un sujet de conférence, et comment avez-vous abordé sa préparation ?_**
```

**_Maintenant que l'événement est derrière nous, que retenez-vous de cette expérience ?_**

**Sofia**: C'était une très belle expérience et les échanges que j'ai pu avoir suite à ma conférence ont été très intéressants. C'est très enrichissant d'échanger avec la communauté et de voir que des pratiques similaires ont lieu ailleurs et pouvoir les enrichir dans les deux sens.
![Photo de Benoit](/images/posts/forumphp2021/benoit_speaker.jpg)
**Benoit**: C'était un vrai plaisir de pouvoir échanger avec de vraies personnes, sans écrans interposés ! Côté speaker, l'organisation était au top et j'ai eu pleins d'échanges prometteurs sur le potentiel des Fibers. Côté conférences, j'ai vu plein de choses intéressantes, ça donne toujours matière à réfléchir, que l'on partage le point de vue exposé ou non. Merci encore à l'AFUP pour avoir mis toute cette énergie au service d'un si bel événement.

> Le forum, particulièrement cette édition en présentiel, c’est retrouver toute une communauté qui partage la même passion. Encore merci aux conférencières et conférenciers, merci aux organisatrices et organisateurs et à l’année prochaine.

Reviewer 3: ```suggestion
Le forum, particulièrement cette édition en présentiel, c’est retrouver toute une communauté qui partage la même passion. Encore merci aux conférencières et conférenciers, merci aux organisatrices et organisateurs… Et à l’année prochaine !
```

> XOXO à l’année prochaine. 🐭

Reviewer 3: Cette phrase sonne comme une conclusion… Alors qu'il y en a déjà une à la phrase au dessus. C'est confusant ^^
